### DISCUSSION TITLE
Call matlab code in gadgetron with execute name="file_name" type="matlab" instead of the port number

### DISCUSSION THREAD

**Discussion by gothamninja** [gadgetron/gadgetron#1241] [Q&A]
Hi all, 

I'm now focusing on engaging an off-line reconstruction algorithm (written in MATLAB) with gadgetron. At the start, I debugged all my code with the debug mode (jump in matlab and use breakpoint in matlab) - in the config file I use
`<external>
      <connect port="23200"/>
 </external>`
and in matlab I use (handle_connection_disorder.m is the main MATLAB function file): 
`gadgetron.external.listen(23200,@gadgetron_moco_siemensgre.handle_connection_disorder)`
And it could work well.

However, then I would like to avoid to use command in matlab and would like to do the whole routine automatically, therefore, I replaced the <external> part as the following code: 
`<external>
      <execute name="gadgetron.gadgetron_moco_siemensgre.handle_connection_disorder" type="matlab"/>
    </external> `
And put the folder that contains all of the supporting code in the path: MATLAB Add-Ons/Toolboxes/gadgetron/+gadgetron/+gadgetron_moco_siemensgre/Function

And when I run the code, it could get in the main function (handle_connection_disorder), but could not get into the first 
support function (bufferData2Rec_double_twix_header.m - saved as an independent function m file in the path above) in the main function. The error message on the gadgetron side goes as:

`Undefined function 'bufferData2Rec_double_twix_header' for input arguments of type 'gadgetron.types.ReconData'.
Error in gadgetron.gadgetron_moco_siemensgre.handle_connction_disorder (line 24)
rec = bufferData2Rec_double_twix_header(bufferData, connection);
Error in gadgetron.external.main (line 18)
              feval(module, connection);`

I know I am not following the very typical way to deal with the gadgetron matlab code, because there are too many support matlab functions in my current code. I wonder if there is an easy way to solve this problem without modifying the matlab codes too much. I've attached a picture to show the folder path that I set and the main matlab function code just in case. 

Many thanks for having a look into this!

[matlab_function_config_file.zip](https://github.com/gadgetron/gadgetron/files/13320906/matlab_function_config_file.zip)

**Comment by rajramasawmy** [comment]
In linux, I set this environment variable:
``` bash
export MATLAB_USE_USERWORK=1
```
Matlab then opens to the Documents/MATLAB folder in my home directory - where I have a pathdef.m with all the files I need. That has worked for me for calling matlab from the gadgetron
